File: src/WePrint.Common/Models/AddressModel.cs
namespace WePrint.Common.Models
{
    public class AddressModel
    {
        public string StreetAddress { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public int ZipCode { get; set; }

        public AddressModel GetPublicAddress()
        {
            AddressModel returnable = this;
            returnable.StreetAddress = "";
            return returnable;
        }
    }
}

File: src/WePrint/ClientApp/src/api/AddressModel.js
export class AddressModel {

    constructor(streetAddress, city, state, zipCode){
        this.streetAdress = streetAddress;
        this.city = city;
        this.state = state;
        this.zipCode = zipCode;
    }

    static Equals(a, b)
    {
        if(a === b && a === null) return true;
        if(a !== null || b!== null) return false;
        if(a === b && a === undefined) return true;
        if(a !== undefined || b !== undefined) return false;
        return a.streetAdress === b.streetAdress && 
            a.city === b.city &&
            a.state === b.state &&
            a.zipCode == b.zipCode;
    }
}